contrib/Notepad++_2_Julia.ahk

```diff
@@ -14,61 +51,134 @@ OpenJulia()
   Process, Exist, %JuliaPID%
   if %ErrorLevel% = 0
   {
-    Run julia.bat,,,JuliaPID
+    Run julia.exe,,,JuliaPID
     hasRunJulia = 1
     ReturnToNpp()
     return
   }
 }
-
+;---------------------------
 ReOpenJulia()
 {
   WinClose, ahk_pid %JuliaPID%
   OpenJulia()
 }
-
+;---------------------------
 SendToJulia()
 {
   OpenJulia()
   WinWait, ahk_pid %JuliaPID%
   WinActivate, ahk_pid %JuliaPID%
   WinWaitActive, ahk_pid %JuliaPID%
-  Send {Alt Down}{Space}{Alt up}ep{Enter}
-}
 
+  ; Past via menu (must disable SendMode Input)
+  ;Send {Alt Down}{Space}{Alt up}ep{Enter}
+
+  ; Past with send clipboard 
+  StringReplace clipboard2, clipboard, \r\n, \n, All
+  SendInput {Raw}%clipboard2%\n
+}
+;---------------------------
 ReturnToNpp()
 {
   WinActivate, ahk_pid %NppPID%
 }
-  
-
+;---------------------------
 PassLine()
 {
   WinGetCLASS, currentClass, A
   If currentClass = Notepad++
   { 
     WinGet, NppPID, PID, A
-    send {Home}{Shift Down}{End}{Shift Up}{Ctrl Down}c{Ctrl Up}{End}
+    SendInput {Home}{Shift Down}{End}{Shift Up}{Ctrl Down}c{Ctrl Up}{End}
     SendToJulia()
     ReturnToNpp()
   }
 }
-
+;---------------------------
 PassBlock()
 {
   WinGetCLASS, currentClass, A
   If currentClass = Notepad++
   {
     WinGet, NppPID, PID, A
-    send {Ctrl Down}c{Ctrl Up}
+    SendInput {Ctrl Down}c{Ctrl Up}
     SendToJulia()
     ReturnToNpp()
   }
 }
-  
-
+;---------------------------
+; Eval current line (Left Shift - Enter)
 LShift & Enter:: PassLine()
+;---------------------------
+; Eval selected block (Right Shift - Enter)
+RShift & Enter:: PassBlock()
+;---------------------------
+; Run Julia (Win-F12)
+#$F12:: ReOpenJulia()
+
+;================================================
+; REPL enhancements
+; Work only on console window
+#IfWinActive ahk_class ConsoleWindowClass
+
+;=============================
+; Scroll command window back and forward
+; Ctrl+PageUp / PageDown
+;=============================
+^PgUp:: SendInput {WheelUp}
+
+^PgDn:: SendInput {WheelDown}
+
+;=============================
+; Paste in REPL
+; Ctrl-V
+; Better version
+;=============================
+^V::
+StringReplace clipboard2, clipboard, \r\n, \n, All
+SendInput {Raw}%clipboard2%
+return
+
+;=============================
+; Paste in REPL
+; Ctrl-V
+; Old version (must disable SendMode Input)
+;=============================
+;^V::
+; English menu (Edit->Paste)
+;Send !{Space}ep
+;return
 
-Ctrl & d:: PassBlock()
+#IfWinActive
+;================================================
 
-$F3:: ReOpenJulia()
+;=============================
+; Greek Letters (lowercase)
+; Control-Win-<char>
+; char 'j' is free...
+;=============================
+^#a:: SendInput {U+03B1} ; alpha
```

Comment:
Since this (and the others below) are generic, not specific to Notepad++. Perhaps these should be a separate AHK script?